shell/platform/android/io/flutter/plugin/platform/PlatformViewsController.java

```diff
@@ -380,33 +545,47 @@ public void setDirection(int viewId, int direction) {
                     + viewId
                     + ")");
           }
-          final PlatformView platformView = platformViews.get(viewId);
-          if (platformView == null) {
-            Log.e(TAG, "Setting direction to an unknown view with id: " + viewId);
-            return;
+
+          View embeddedView;
+
+          if (usesVirtualDisplay(viewId)) {
+            final VirtualDisplayController controller = vdControllers.get(viewId);
+            embeddedView = controller.getView();
+          } else {
+            final PlatformView platformView = platformViews.get(viewId);
+            if (platformView == null) {
+              Log.e(TAG, "Setting direction to an unknown view with id: " + viewId);
+              return;
+            }
+            embeddedView = platformView.getView();
           }
-          ensureValidAndroidVersion(Build.VERSION_CODES.KITKAT_WATCH);
-          final View view = platformView.getView();
-          if (view == null) {
+          if (embeddedView == null) {
             Log.e(TAG, "Setting direction to a null view with id: " + viewId);
             return;
           }
-          view.setLayoutDirection(direction);
+          embeddedView.setLayoutDirection(direction);
         }
 
         @Override
         public void clearFocus(int viewId) {
-          final PlatformView platformView = platformViews.get(viewId);
-          if (platformView == null) {
-            Log.e(TAG, "Clearing focus on an unknown view with id: " + viewId);
-            return;
+          View embeddedView;
+
+          if (usesVirtualDisplay(viewId)) {
+            final VirtualDisplayController controller = vdControllers.get(viewId);
+            embeddedView = controller.getView();
+          } else {
+            final PlatformView platformView = platformViews.get(viewId);
+            if (platformView == null) {
+              Log.e(TAG, "Clearing focus on an unknown view with id: " + viewId);
```

Comment:
here and elsewhere: These logs seem to violate the style guide - they seem to be debugging information that might be useful to an engine developer but don't really mean a whole lot to a developer. They should be made actionable to end developers (did a plugin forget to make an important API call here?) or removed.

Reply:
Filed https://github.com/flutter/flutter/issues/106444